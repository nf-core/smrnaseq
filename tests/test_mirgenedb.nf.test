
nextflow_pipeline {

    name "Test Workflow main.nf - test_mirgenedb"
    script "main.nf"
    profile "test_mirgenedb"
    tag "mirgenedb"
    tag "pipeline"

    test("test_mirgenedb") {

        when {
            params {
                outdir = "$outputDir"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(UTILS.removeNextflowVersion("$outputDir")).match("software_versions") },
                { assert workflow.trace.succeeded().size() == 90 },
                { assert workflow.trace.failed().size() == 1 },

                { assert snapshot(
                    path("$outputDir/mirna_quant/bam/mature/Clone1_N1_mature.sorted.flagstat").exists(),
                    path("$outputDir/mirna_quant/bam/mature/Clone1_N1_mature.sorted.stats").exists(),
                    path("$outputDir/mirna_quant/bam/mature/Clone9_N1_mature.sorted.flagstat"),
                    path("$outputDir/mirna_quant/bam/mature/Clone9_N1_mature.sorted.idxstats"),
                    path("$outputDir/mirna_quant/bam/mature/Control_N1_mature.sorted.flagstat"),
                    path("$outputDir/mirna_quant/bam/mature/Control_N1_mature.sorted.stats"),
                    path("$outputDir/mirna_quant/bam/mature/Clone9_N1_mature.sorted.stats"),
                    path("$outputDir/mirna_quant/bam/mature/Control_N1_mature.sorted.idxstats"),
                    path("$outputDir/mirna_quant/bam/mature/Clone1_N1_mature.sorted.idxstats").exists(),
                    path("$outputDir/mirna_quant/bam/hairpin/Control_N1_mature_hairpin.sorted.idxstats"),
                    path("$outputDir/mirna_quant/bam/hairpin/Control_N1_mature_hairpin.sorted.flagstat"),
                    path("$outputDir/mirna_quant/bam/hairpin/Clone9_N1_mature_hairpin.sorted.idxstats"),
                    path("$outputDir/mirna_quant/bam/hairpin/Clone9_N1_mature_hairpin.sorted.stats"),
                    path("$outputDir/mirna_quant/bam/hairpin/Clone1_N1_mature_hairpin.sorted.idxstats").exists(),
                    path("$outputDir/mirna_quant/bam/hairpin/Clone1_N1_mature_hairpin.sorted.flagstat").exists(),
                    path("$outputDir/mirna_quant/bam/hairpin/Clone1_N1_mature_hairpin.sorted.stats").exists(),
                    path("$outputDir/mirna_quant/bam/hairpin/Control_N1_mature_hairpin.sorted.stats"),
                    path("$outputDir/mirna_quant/bam/hairpin/Clone9_N1_mature_hairpin.sorted.flagstat")
                ).match("mirna_quant_bam") },

                { assert snapshot(
                    path("$outputDir/mirna_quant/edger_qc/hairpin_edgeR_MDS_distance_matrix.txt").exists(),
                    path("$outputDir/mirna_quant/edger_qc/hairpin_counts.csv").exists(),
                    path("$outputDir/mirna_quant/edger_qc/hairpin_edgeR_MDS_plot_coordinates.txt").exists(),
                    path("$outputDir/mirna_quant/edger_qc/hairpin_log2CPM_sample_distances.txt").exists(),
                    path("$outputDir/mirna_quant/edger_qc/hairpin_logtpm.txt").exists(),
                    path("$outputDir/mirna_quant/edger_qc/hairpin_normalized_CPM.txt").exists(),
                    path("$outputDir/mirna_quant/edger_qc/hairpin_logtpm.csv").exists(),
                    path("$outputDir/mirna_quant/edger_qc/mature_counts.csv").exists(),
                    path("$outputDir/mirna_quant/edger_qc/mature_edgeR_MDS_distance_matrix.txt").exists(),
                    path("$outputDir/mirna_quant/edger_qc/hairpin_unmapped_read_counts.txt").exists(),
                    path("$outputDir/mirna_quant/edger_qc/mature_edgeR_MDS_plot_coordinates.txt").exists(),
                    path("$outputDir/mirna_quant/edger_qc/mature_log2CPM_sample_distances.txt").exists(),
                    path("$outputDir/mirna_quant/edger_qc/mature_logtpm.csv").exists(),
                    path("$outputDir/mirna_quant/edger_qc/mature_logtpm.txt").exists(),
                    path("$outputDir/mirna_quant/edger_qc/mature_normalized_CPM.txt").exists(),
                    path("$outputDir/mirna_quant/edger_qc/mature_unmapped_read_counts.txt").exists()
                ).match("mirna_quant_edger_qc") },

                { assert snapshot(
                    path("$outputDir/genome_quant/bam/Clone9_N1_mature_hairpin_genome.sorted.stats"),
                    path("$outputDir/genome_quant/bam/Clone9_N1_mature_hairpin_genome.sorted.idxstats"),
                    path("$outputDir/genome_quant/bam/Clone1_N1_mature_hairpin_genome.sorted.stats"),
                    path("$outputDir/genome_quant/bam/Control_N1_mature_hairpin_genome.sorted.stats"),
                    path("$outputDir/genome_quant/bam/Clone1_N1_mature_hairpin_genome.sorted.flagstat"),
                    path("$outputDir/genome_quant/bam/Clone9_N1_mature_hairpin_genome.sorted.flagstat"),
                    path("$outputDir/genome_quant/bam/Clone1_N1_mature_hairpin_genome.sorted.idxstats"),
                    path("$outputDir/genome_quant/bam/Control_N1_mature_hairpin_genome.sorted.idxstats"),
                    path("$outputDir/genome_quant/bam/Control_N1_mature_hairpin_genome.sorted.flagstat")
                ).match("genome_quant_bam") },

                { assert snapshot(
                    path("$outputDir/mirdeep2/result_Clone1_N1.csv").exists(),
                    path("$outputDir/mirdeep2/result_Control_N1.csv").exists(),
                    path("$outputDir/mirdeep2/result_Control_N1.bed").exists(),
                    path("$outputDir/mirdeep2/result_Control_N1.bed").exists(),
                    path("$outputDir/mirdeep2/result_Control_N1.html").exists(),
                    path("$outputDir/mirdeep2/result_Control_N1.html").exists()
                ).match("mirdeep2") },

                { assert snapshot(
                    path("$outputDir/multiqc/multiqc_report.html").exists()
                ).match("multiqc") },

                { assert snapshot(
                    path("$outputDir/multiqc/multiqc_data/fastqc-status-check-heatmap.txt"),
                    path("$outputDir/multiqc/multiqc_data/fastp_filtered_reads_plot.txt"),
                    path("$outputDir/multiqc/multiqc_data/fastqc_overrepresented_sequences_plot.txt"),
                    path("$outputDir/multiqc/multiqc_data/fastqc-1_top_overrepresented_sequences_table.txt").exists(),
                    path("$outputDir/multiqc/multiqc_data/fastqc-1_sequence_counts_plot.txt"),
                    path("$outputDir/multiqc/multiqc_data/fastqc-1_per_sequence_gc_content_plot_Percentages.txt"),
                    path("$outputDir/multiqc/multiqc_data/multiqc_citations.txt"),
                    path("$outputDir/multiqc/multiqc_data/samtools-stats-dp.txt"),
                    path("$outputDir/multiqc/multiqc_data/fastqc_sequence_length_distribution_plot.txt"),
                    path("$outputDir/multiqc/multiqc_data/fastp-seq-content-n-plot_Read_1_Before_filtering.txt"),
                    path("$outputDir/multiqc/multiqc_data/fastqc-1_sequence_duplication_levels_plot.txt"),
                    path("$outputDir/multiqc/multiqc_data/fastqc-1_per_base_sequence_quality_plot.txt"),
                    path("$outputDir/multiqc/multiqc_data/multiqc_general_stats.txt"),
                    path("$outputDir/multiqc/multiqc_data/fastqc-1_per_base_n_content_plot.txt"),
                    path("$outputDir/multiqc/multiqc_data/fastqc_per_base_n_content_plot.txt"),
                    path("$outputDir/multiqc/multiqc_data/fastp-seq-quality-plot_Read_1_After_filtering.txt"),
                    path("$outputDir/multiqc/multiqc_data/fastqc_per_sequence_quality_scores_plot.txt"),
                    path("$outputDir/multiqc/multiqc_data/fastqc-1_per_sequence_quality_scores_plot.txt"),
                    path("$outputDir/multiqc/multiqc_data/fastqc_top_overrepresented_sequences_table.txt").exists(),
                    path("$outputDir/multiqc/multiqc_data/fastqc-1_overrepresented_sequences_plot.txt"),
                    path("$outputDir/multiqc/multiqc_data/fastqc-1-status-check-heatmap.txt"),
                    path("$outputDir/multiqc/multiqc_data/fastqc_sequence_counts_plot.txt"),
                    path("$outputDir/multiqc/multiqc_data/fastp-seq-quality-plot_Read_1_Before_filtering.txt"),
                    path("$outputDir/multiqc/multiqc_data/samtools_alignment_plot.txt"),
                    path("$outputDir/multiqc/multiqc_data/fastqc_per_base_sequence_quality_plot.txt"),
                    path("$outputDir/multiqc/multiqc_data/fastp-seq-content-n-plot_Read_1_After_filtering.txt"),
                    path("$outputDir/multiqc/multiqc_data/fastqc_adapter_content_plot.txt"),
                    path("$outputDir/multiqc/multiqc_data/fastqc_sequence_duplication_levels_plot.txt"),
                    path("$outputDir/multiqc/multiqc_data/fastqc_per_sequence_gc_content_plot_Percentages.txt"),
                    path("$outputDir/multiqc/multiqc_data/fastqc-1_per_sequence_gc_content_plot_Counts.txt"),
                    path("$outputDir/multiqc/multiqc_data/fastqc-1_adapter_content_plot.txt"),
                    path("$outputDir/multiqc/multiqc_data/fastp-seq-content-gc-plot_Read_1_Before_filtering.txt"),
                    path("$outputDir/multiqc/multiqc_data/fastqc_per_sequence_gc_content_plot_Counts.txt"),
                    path("$outputDir/multiqc/multiqc_data/multiqc_sources.txt").exists(),
                    path("$outputDir/multiqc/multiqc_data/fastp-seq-content-gc-plot_Read_1_After_filtering.txt")
                ).match("multiqc_multiqc_data")}
            )
        }
    }

}
