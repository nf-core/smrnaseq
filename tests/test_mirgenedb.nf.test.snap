{
    "genome_quant_bam": {
        "content": [
            "Clone9_N1_mature_hairpin_genome.sorted.stats:md5,201486e8c88b0dbb639776f9dacefe66",
            "Clone9_N1_mature_hairpin_genome.sorted.idxstats:md5,aa37c5da7c2b4505ce58c3a21f97121c",
            "Clone1_N1_mature_hairpin_genome.sorted.stats:md5,5679af445f4628e0f4137f339319c950",
            "Control_N1_mature_hairpin_genome.sorted.stats:md5,4da4f877f06083f616a0d76edd04ea22",
            "Clone1_N1_mature_hairpin_genome.sorted.flagstat:md5,5bb521c495f1c450835299b1eb88dc84",
            "Clone9_N1_mature_hairpin_genome.sorted.flagstat:md5,6a8ad3be2ca0fa924fd32a04293d4ce4",
            "Clone1_N1_mature_hairpin_genome.sorted.idxstats:md5,d92f9eae7657418858e6d2b69436f74f",
            "Control_N1_mature_hairpin_genome.sorted.idxstats:md5,a11f543771cea6b383fb596f60e998c3",
            "Control_N1_mature_hairpin_genome.sorted.flagstat:md5,df2a57ac3b36f5d40793d3105a4bb2d1"
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-20T13:36:32.067039888"
    },
    "software_versions": {
        "content": [
            "{BOWTIE_MAP_GENOME={bowtie=1.3.0, samtools=1.16.1}, BOWTIE_MAP_HAIRPIN={bowtie=1.3.0, samtools=1.16.1}, BOWTIE_MAP_MATURE={bowtie=1.3.0, samtools=1.16.1}, BOWTIE_MAP_SEQCLUSTER={bowtie=1.3.0, samtools=1.16.1}, FASTP={fastp=0.23.4}, FASTQC_RAW={fastqc=0.12.1}, FASTQC_TRIM={fastqc=0.12.1}, FORMAT_HAIRPIN={fastx_toolkit=0.0.14}, FORMAT_MATURE={fastx_toolkit=0.0.14}, INDEX_HAIRPIN={bowtie=1.3.0}, INDEX_MATURE={bowtie=1.3.0}, MIRDEEP2_MAPPER={mirdeep2=2.0.1}, MIRDEEP2_MIRDEEP2={mirdeep2=2.0.1}, PARSE_HAIRPIN={seqkit=2.6.1}, PARSE_MATURE={seqkit=2.6.1}, SAMTOOLS_FLAGSTAT={samtools=1.21}, SAMTOOLS_IDXSTATS={samtools=1.21}, SAMTOOLS_INDEX={samtools=1.21}, SAMTOOLS_SORT={samtools=1.21}, SAMTOOLS_STATS={samtools=1.21}, SEQCLUSTER_COLLAPSE={seqcluster=1.2.9}, SEQKIT_FQ2FA={seqkit=2.8.0}, SEQKIT_REPLACE={seqkit=2.8.0}, Workflow={nf-core/smrnaseq=v2.3.2dev}}"
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-01T20:13:41.274631809"
    },
    "mirna_quant_bam": {
        "content": [
            true,
            true,
            "Clone9_N1_mature.sorted.flagstat:md5,adf40ba27907b6ef726d6c5923a731b9",
            "Clone9_N1_mature.sorted.idxstats:md5,8302f401476f5c8fee3333e1c742c05e",
            "Control_N1_mature.sorted.flagstat:md5,f8df7690d20014518f47dc2fe39debec",
            "Control_N1_mature.sorted.stats:md5,35a03b446785dbf240d755ea9edac0f6",
            "Clone9_N1_mature.sorted.stats:md5,d2cb555133269b4503e4eb490dd81b3e",
            "Control_N1_mature.sorted.idxstats:md5,b7a382b1d0f5cba6cb94b3b5a6b18f84",
            true,
            "Control_N1_mature_hairpin.sorted.idxstats:md5,79dc5e82ff88e7379c893549224cd87f",
            "Control_N1_mature_hairpin.sorted.flagstat:md5,1dc7b98f0014a99a20de7c09a6b95340",
            "Clone9_N1_mature_hairpin.sorted.idxstats:md5,f3ed5bf23f73d41c42d3da0bf30f89ea",
            "Clone9_N1_mature_hairpin.sorted.stats:md5,9fda01757030fe96a60171965117b2e4",
            true,
            true,
            true,
            "Control_N1_mature_hairpin.sorted.stats:md5,a96b6d37b444f52c66a3dac9d7de0f36",
            "Clone9_N1_mature_hairpin.sorted.flagstat:md5,678f4f9e98c3e1fcc5af54e8dd06fbbc"
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-19T15:14:06.231457002"
    },
    "mirdeep2": {
        "content": [
            true,
            true,
            true,
            true,
            true,
            true
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-26T18:15:04.45050483"
    },
    "mirna_quant_edger_qc": {
        "content": [
            true,
            true,
            true,
            true,
            true,
            true,
            true,
            true,
            true,
            true,
            true,
            true,
            true,
            true,
            true,
            true
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "23.10.0"
        },
        "timestamp": "2024-08-30T20:30:51.020931866"
    },
    "multiqc_multiqc_data": {
        "content": [
            "fastqc-status-check-heatmap.txt:md5,3f897028847a4a3a5f666325cd732067",
            "fastp_filtered_reads_plot.txt:md5,277be8c15684b9f09737bc13bcd3a055",
            "fastqc_overrepresented_sequences_plot.txt:md5,7d7ecccd5d6c7fd63439785888c0c164",
            true,
            "fastqc-1_sequence_counts_plot.txt:md5,59faee895ea86c12a4124d417e3bbd63",
            "fastqc-1_per_sequence_gc_content_plot_Percentages.txt:md5,0a4b4285f2c53dca216c107decc9921f",
            "multiqc_citations.txt:md5,57db2426be011862828d18f767d25b57",
            "samtools-stats-dp.txt:md5,45c0315bade3f07942ded1ead37c1489",
            "fastqc_sequence_length_distribution_plot.txt:md5,8b5cf1e3429a1ea0b3c63cfb176e1014",
            "fastp-seq-content-n-plot_Read_1_Before_filtering.txt:md5,a0502dd4f701c9deb646ffbec80c09de",
            "fastqc-1_sequence_duplication_levels_plot.txt:md5,2072cda513c8884047d9d11c8aacbf33",
            "fastqc-1_per_base_sequence_quality_plot.txt:md5,cafad80f4e07df53590cbabbbd024629",
            "multiqc_general_stats.txt:md5,950d3fb06c211e984084e6de9dad6bb3",
            "fastqc-1_per_base_n_content_plot.txt:md5,a0502dd4f701c9deb646ffbec80c09de",
            "fastqc_per_base_n_content_plot.txt:md5,d907ac1ac9a4f19908b7b025eb75abfe",
            "fastp-seq-quality-plot_Read_1_After_filtering.txt:md5,0742b9813dcc95d4c62c52c83dec390c",
            "fastqc_per_sequence_quality_scores_plot.txt:md5,c7aacf1ab75fbe89f86e33273aefaf26",
            "fastqc-1_per_sequence_quality_scores_plot.txt:md5,6b8b8ddf52e9dcc22a4ac00c99105301",
            true,
            "fastqc-1_overrepresented_sequences_plot.txt:md5,e2e8f71f05beec9a8a7809e47ae738ad",
            "fastqc-1-status-check-heatmap.txt:md5,d9c3ce24536a948e1fe9b84c55421ab7",
            "fastqc_sequence_counts_plot.txt:md5,4861f0dc120e57e0359c53f417756b0c",
            "fastp-seq-quality-plot_Read_1_Before_filtering.txt:md5,e9d8e3289f84f5a1ae6775813ec5a9b4",
            "samtools_alignment_plot.txt:md5,b841ffce110bde994ccc6e977d2f856e",
            "fastqc_per_base_sequence_quality_plot.txt:md5,a8adbff96d9adb317079e6becd7a80f6",
            "fastp-seq-content-n-plot_Read_1_After_filtering.txt:md5,a0502dd4f701c9deb646ffbec80c09de",
            "fastqc_adapter_content_plot.txt:md5,bd0fdc9c856c55598976b5a46c23a677",
            "fastqc_sequence_duplication_levels_plot.txt:md5,2b1cbdce195d2aedc4bff4c5e9b618d4",
            "fastqc_per_sequence_gc_content_plot_Percentages.txt:md5,6906cedd750c0d43a26fdcddeacce257",
            "fastqc-1_per_sequence_gc_content_plot_Counts.txt:md5,96abe7c73fc433142fbdbacb1e67e87f",
            "fastqc-1_adapter_content_plot.txt:md5,c9d77edf35d9afb8b6e86b939b5be596",
            "fastp-seq-content-gc-plot_Read_1_Before_filtering.txt:md5,910576f4999a406ea37306b8dc4eb45b",
            "fastqc_per_sequence_gc_content_plot_Counts.txt:md5,7cdf079a279cf080f2e2d7ab00b4f134",
            true,
            "fastp-seq-content-gc-plot_Read_1_After_filtering.txt:md5,585ec288b2514de54e8fb6251d1e0f98"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-19T03:58:28.495279269"
    },
    "multiqc": {
        "content": [
            true
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "23.10.0"
        },
        "timestamp": "2024-08-30T20:30:51.144222162"
    }
}